libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs

```diff
@@ -39,6 +39,8 @@ public class LuisRecognizer : ITelemetryRecognizer
         /// </summary>
         public const string LuisTraceLabel = "Luis Trace";
 
+        private const string CacheKey = "LuisResult";
```

Comment:
Using TurnState as the cache is smart, i like that. 

What i'm not sure about is using a fixed cache key. If i have two dialogs both with luis recoginzer, won't the result be conflicted? 

Please test against the case that multiple luis recoginzers are called in one turn, like one dialog is begin another dialog, or one utterance are bubbled up more than once.